packages/drawer/src/ux-modal-service.ts

```diff
@@ -169,7 +167,33 @@ export class ModalService {
     return element;
   }
 
-  public open(options: ModalServiceOptions): UxDrawer & ModalServiceDrawer {
+  private createCompositionContext(
+    container: any, 
+    host: Element, 
+    bindingContext: ModalBindingContext, 
+    settings: {model?: any, view?: any, viewModel?: any},
+    slot? : ViewSlot
+    ): CompositionContext {
+    return {
+      container,
+      bindingContext: settings.viewModel ? null : bindingContext,
+      viewResources: null as any,
+      model: settings.model,
+      view: settings.view,
+      viewModel: settings.viewModel,
+      viewSlot: slot || new ViewSlot(host, true),
+      host
+    };
+  }
+
+  private ensureViewModel(compositionContext: CompositionContext): Promise<CompositionContext> {
+    if (typeof compositionContext.viewModel === 'object') {
+      return Promise.resolve(compositionContext);
+    }
```

Comment:
lets experiment with this:
```suggestion
    const viewModel = compositionContext.viewModel;
    if (typeof viewModel === 'object') {
      // emulate an anonymous class that always return the view model object above
      compositionContext.viewModel = function() {
        return viewModel;
      };
    }
```
And then we always pass the composition context through the composition engine

Reply:
I have tried this experiment. However this does not fix our composition issue yet. It should be noted that the composition issue happened not only for when we pass on a `VM` class but also when we pass a `VM` path or a view only. 

I have updated the [Dumber Gist](https://gist.dumber.app/?gist=938f48392ae45cfeb6d77cef5ab5084c&open=src%2Fapp.ts&open=src%2Fapp.html) to reflect this issue with View Only composition.